docs/software-architecture-and-design.md

```diff
@@ -98,44 +99,43 @@ The Field class represents the state of the physical field being played on, whic
 ### GameState
 These represent the current state of the game as dictated by the Gamecontroller. These provide functions like `isPlaying()`, `isHalted()` which tell the rest of the system what game state we are in, and make decisions accordingly. We need to obey the rules!
 
-
-## Primitives
-Primitives are very simple actions of things a robot can do. It does not represent or include _how_ these things are done. Some examples are:
-* Moving in a straight line to a position
-* Pivoting around a point
-* Kicking the ball at a certain direction
-
-Primitives act as the abstraction between our AI, and our robot firmware. It's much easier for our AI to send a `Primitive` to a robot telling it what it wants it to do, and have the robot responsible for making sure it does what it's told. For every `Primitive` in our `AI` software, there is an equivalent `Primitive` implementation in our robot firmware. When robots receive a `Primitive` command, they perform their own logic and control in order to perform the task specified by the `Primitive`.
-
-
 ## Intents
-An `Intent` represents a simple thing the `AI` wants (or intends for) a robot to do. It does not represent or include _how_ these things are achieved. Some examples are:
+An `Intent` represents a simple thing the [AI](#ai) wants (or intends for) a robot to do. It does not represent or include _how_ these things are achieved. Some examples are:
 * Moving to a position (without colliding with anything on its way)
 * Pivoting around a point
 * Kicking the ball at a certain direction or at a target
 
-Intents are very similar to Primitives, but include slightly more logic. `Intents` can include extra parameters or data that `Primitives` do not, such as how much to avoid the ball by while moving. In this way, `Intents` are more "context-aware" than `Primitives`, and represent slightly higher-level commands.
-
+There are two types of `Intent`s: `DirectPrimitiveIntent`s and `NavigatingIntent`s. `DirectPrimitiveIntent`s directly represent the [Primitives](#primitives) that the AI is trying to send to the the robots. `NavigatingIntent`s are intents that require moving while avoiding obstacles, so they contain extra parameters to help with [Navigation](#navigation).
```

Comment:
"... that the AI is trying to send to the the robots" should only have one "the".

Reply:
nice